zerobuf/data/Progress.cpp

```diff
@@ -0,0 +1,53 @@
+
+/* Copyright (c) 2016, Human Brain Project
+ *                     Stefan.Eilemann@epfl.ch
+ */
+
+#include "Progress.h"
+
+namespace zerobuf
+{
+namespace data
+{
+Progress::Progress( const unsigned long expected )
+    : _expected( expected )
+    , _count( 0 )
+{
+    _update();
+}
+
+Progress::Progress( const std::string& operation, unsigned long expected )
+    : detail::Progress( operation, 0.f )
+    , _expected( expected )
+    , _count( 0 )
+{
+    _update();
+}
+
+void Progress::restart( const unsigned long expected )
+{
+    _expected = expected;
+    _count = 0;
+    _update();
+}
+
+unsigned long Progress::operator+=( unsigned long inc )
```

Comment:
One is making negative progress? C&P of the boost::progress_display API, which does not have this.